src/components/WineLister.jsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import axios from 'axios';
+import { useWineList } from '../contexts/WineListContext';
+import WineBottle from './WineBottle';
+
+function WineLister() {
+  const url = 'http://localhost:8000/bottles';
+  const { wineList, setWineList } = useWineList();
+  React.useEffect(() => {
+    axios.get(url)
+      .then((response) => (setWineList(response.data)));
+  }, [wineList]);
```

Comment:
the effect doesn't depend on wineList, so you can remove it from the array.

since you have a context, do the useEffect in the provider ;)

Reply:
Fixed :)